### PR TITLE
[LAB-8] CRUD

### PR DESCRIPTION
**Pull request by Marko-Shevchuk**

### CODE REVIEW

File: src/main/java/com/pnu/lab/control/labcontrol/api/AssignmentCommentController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        return service.update(assignmentComment);
+    }
+
+    @PostMapping("/list")
+    public List<AssignmentComment> getList(@Valid @RequestBody SearchRequest searchRequest) {
+        return service.getList(searchRequest);
+
+    }
+
+    @GetMapping("/{id}")
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/api/AssignmentCommentController.java:35]: useless

File: src/main/java/com/pnu/lab/control/labcontrol/api/SubmissionCommentController.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return service.update(submissionComment);
+    }
+
+    @PostMapping("/list")
+    public List<SubmissionComment> getList(@Valid @RequestBody SearchRequest searchRequest) {
+        return service.getList(searchRequest);
+
+    }
+
+    @GetMapping("/{id}")
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/api/SubmissionCommentController.java:35]: +1

File: src/main/java/com/pnu/lab/control/labcontrol/repository/AssignmentRepository.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.pnu.lab.control.labcontrol.domain.Assignment;
+import com.pnu.lab.control.labcontrol.domain.QAssignment;
+import com.querydsl.core.types.dsl.EntityPathBase;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface AssignmentRepository extends BaseSearchRepository<Assignment> {
+    QAssignment qAssignment = QAssignment.assignment;
+    @Override
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/repository/AssignmentRepository.java:11]: space

File: src/main/java/com/pnu/lab/control/labcontrol/service/AssignmentCommentService.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import com.pnu.lab.control.labcontrol.repository.BaseSearchRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AssignmentCommentService extends AbstractSearchService<AssignmentComment> {
+
+    private final AssignmentCommentRepository repository;
+
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/service/AssignmentCommentService.java:14]: override ```delete(AssignmentComment comment)```, set ```deleted``` to ```true``` and then just ```update```

File: src/main/java/com/pnu/lab/control/labcontrol/api/AssignmentController.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    @PutMapping
+    public Assignment update(@Valid @RequestBody Assignment assignment) {
+        return service.update(assignment);
+    }
+
+    @PostMapping("/list")
+    public List<Assignment> getList(@Valid @RequestBody SearchRequest searchRequest) {
+        return service.getList(searchRequest);
+
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/api/AssignmentController.java:39]: remove space

File: src/main/java/com/pnu/lab/control/labcontrol/api/AssignmentCommentController.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+    @PutMapping
+    public AssignmentComment update(@Valid @RequestBody AssignmentComment assignmentComment) {
+        return service.update(assignmentComment);
+    }
+
+    @PostMapping("/list")
+    public List<AssignmentComment> getList(@Valid @RequestBody SearchRequest searchRequest) {
+        return service.getList(searchRequest);
+
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/api/AssignmentCommentController.java:32]: remove space

File: src/main/java/com/pnu/lab/control/labcontrol/api/SubmissionCommentController.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+    @PutMapping
+    public SubmissionComment update(@Valid @RequestBody SubmissionComment submissionComment) {
+        return service.update(submissionComment);
+    }
+
+    @PostMapping("/list")
+    public List<SubmissionComment> getList(@Valid @RequestBody SearchRequest searchRequest) {
+        return service.getList(searchRequest);
+
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/api/SubmissionCommentController.java:32]: +1

File: src/main/java/com/pnu/lab/control/labcontrol/api/SubmissionController.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+    @PutMapping
+    public Submission update(@Valid @RequestBody Submission submission) {
+        return service.update(submission);
+    }
+
+    @PostMapping("/list")
+    public List<Submission> getList(@Valid @RequestBody SearchRequest searchRequest) {
+        return service.getList(searchRequest);
+
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/api/SubmissionController.java:32]: +1

File: src/main/java/com/pnu/lab/control/labcontrol/api/AssignmentCommentController.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    public AssignmentComment update(@Valid @RequestBody AssignmentComment assignmentComment) {
+        return service.update(assignmentComment);
+    }
+
+    @PostMapping("/list")
+    public List<AssignmentComment> getList(@Valid @RequestBody SearchRequest searchRequest) {
+        return service.getList(searchRequest);
+    }
+
+
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/api/AssignmentCommentController.java:34]: remove space

File: src/main/java/com/pnu/lab/control/labcontrol/api/SubmissionCommentController.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    public SubmissionComment update(@Valid @RequestBody SubmissionComment submissionComment) {
+        return service.update(submissionComment);
+    }
+
+    @PostMapping("/list")
+    public List<SubmissionComment> getList(@Valid @RequestBody SearchRequest searchRequest) {
+        return service.getList(searchRequest);
+    }
+
+
```
Comment (ArsenOOO7) [inline src/main/java/com/pnu/lab/control/labcontrol/api/SubmissionCommentController.java:34]: remove empty line

**Review by ArsenOOO7 — APPROVED** [review]

**Review by JJerome-NM — APPROVED** [review]

**Pull request merged**